fastlane/Fastfile

```diff
@@ -8,28 +8,28 @@ end
-  { glotpress: "he",    google_play: "iw-IL",  promo_config: {} },
-  { glotpress: "id",    google_play: "id",     promo_config: {} },
-  { glotpress: "it",    google_play: "it-IT",  promo_config: {} },
-  { glotpress: "ja",    google_play: "ja-JP",  promo_config: {} },
-  { glotpress: "ko",    google_play: "ko-KR",  promo_config: {} },
-  { glotpress: "nl",    google_play: "nl-NL",  promo_config: {} },
-  { glotpress: "pl",    google_play: "pl-PL",  promo_config: {} },
-  { glotpress: "pt-br", google_play: "pt-BR",  promo_config: {} },
-  { glotpress: "ru",    google_play: "ru-RU",  promo_config: {} },
-  { glotpress: "sr",    google_play: "sr",     promo_config: {} },
-  { glotpress: "sv",    google_play: "sv-SE",  promo_config: {} },
-  { glotpress: "th",    google_play: "th",     promo_config: {} },
-  { glotpress: "tr",    google_play: "tr-TR",  promo_config: {} },
-  { glotpress: "vi",    google_play: "vi",     promo_config: {} },
-  { glotpress: "zh-cn", google_play: "zh-CN",  promo_config: {} },
-  { glotpress: "zh-tw", google_play: "zh-TW",  promo_config: {} },
+  { glotpress: "ar", android: "ar",    google_play: "ar",     promo_config: {} },
+  { glotpress: "de", android: "de",    google_play: "de-DE",  promo_config: {} },
+  { glotpress: "en-gb", android: "en-rGB", google_play: "en-US",  promo_config: {} },
+  { glotpress: "es", android: "es",    google_play: "es-ES",  promo_config: {} },
+  { glotpress: "fr", android: "fr-rCA",    google_play: "fr-CA",  promo_config: false },
+  { glotpress: "fr", android: "fr",    google_play: "fr-FR",  promo_config: {} },
+  { glotpress: "he", android: "he",    google_play: "iw-IL",  promo_config: {} },
+  { glotpress: "id", android: "id",    google_play: "id",     promo_config: {} },
+  { glotpress: "it", android: "it",    google_play: "it-IT",  promo_config: {} },
+  { glotpress: "ja", android: "ja",    google_play: "ja-JP",  promo_config: {} },
+  { glotpress: "ko", android: "ko",    google_play: "ko-KR",  promo_config: {} },
+  { glotpress: "nl", android: "nl",    google_play: "nl-NL",  promo_config: {} },
+  { glotpress: "pl", android: "pl",    google_play: "pl-PL",  promo_config: {} },
+  { glotpress: "pt-br", android: "pt-rBR", google_play: "pt-BR",  promo_config: {} },
+  { glotpress: "ru", android: "ru",    google_play: "ru-RU",  promo_config: {} },
+  { glotpress: "sr", android: "sr",    google_play: "sr",     promo_config: {} },
+  { glotpress: "sv", android: "sv",    google_play: "sv-SE",  promo_config: {} },
+  { glotpress: "th", android: "th",    google_play: "th",     promo_config: {} },
+  { glotpress: "tr", android: "tr",    google_play: "tr-TR",  promo_config: {} },
+  { glotpress: "vi", android: "vi",    google_play: "vi",     promo_config: {} },
+  { glotpress: "zh-cn", android: "zh-rCN", google_play: "zh-CN",  promo_config: {} },
+  { glotpress: "zh-tw", android: "zh-rTW", google_play: "zh-TW",  promo_config: {} },
```

Comment:
We have _way_ more than those Mag16 locales in GlotPress for the `strings.xml` files.
We need to download more than just those, so the list that you will pass to `android_download_translations` needs to be expanded to match the full list that is currently defined in `tools/exported-language-codes.xml`.

What I suggest to do is:

 - Rename that constant above to `ALL_LOCALES`
 - Add all the missing entries for locales that are listed in `tools/exported-language-codes.csv` to that `ALL_LOCALES` array, but with no `google_play` key, and with `promo_config: false`.
 - Create a `RELEASE_NOTES_LOCALES = ALL_LOCALES.reject { |h| h[:google_play].nil? }.map { |h| [h[:glotpress], h[:google_play]] }`

```ruby
ALL_LOCALES = [
  # Mag16, which we have release notes for
  { glotpress: "ar", android: "ar",    google_play: "ar",     promo_config: {} },
  { glotpress: "de", android: "de",    google_play: "de-DE",  promo_config: {} },
  { glotpress: "en-gb", android: "en-rGB", google_play: "en-US",  promo_config: {} },
  { glotpress: "es", android: "es",    google_play: "es-ES",  promo_config: {} },
  { glotpress: "fr", android: "fr-rCA",    google_play: "fr-CA",  promo_config: false },
  { glotpress: "fr", android: "fr",    google_play: "fr-FR",  promo_config: {} },
…
  { glotpress: "zh-tw", android: "zh-rTW", google_play: "zh-TW",  promo_config: {} },

  # Additional locales for which we have app strings in GlotPress, but support is unofficial, with no release notes
  { glotpress: "hr", android: "hr", promo_config: false },
  { glotpress: "cy", android: "cy", promo_config: false },
…
]

RELEASE_NOTES_LOCALES = ALL_LOCALES.reject { |h| h[:google_play].nil? }.map { |h| [h[:glotpress], h[:google_play]] }
```

 - Update the call sites for `android_download_translations` to use `ALL_LOCALES` as a parameter for `locales:`
 - Update the call sites for `gp_downloadmetadata` & `android_create_xml_release_notes` (from the `download_metadata_strings` lane) to use `RELEASE_NOTES_LOCALES` instead of `SUPPORTED_LOCALES.map {| hsh | [ hsh[:glotpress], hsh[:google_play] ]}`
 - Update the 3 uses of `SUPPORTED_LOCALES` that are in `ScreenshotFastfile` to use `ALL_LOCALES` instead
 - Delete the old `tools/exported-language-codes.csv` and `tools/update-translations.sh` files

Reply:
Thanks for all the details here! Agree with the feedback and went through with these updates 🙂 Please Let me know if I missed anything!